### PR TITLE
Add support for different tspans in EnsembleGPUArray

### LINKED ISSUE #283: `EnsembleGPUArray` compatibility with different time intervals
Could be fixed with this reparameterization trick:  https://github.com/rtqichen/torchdiffeq/issues/122#issuecomment-738978844

### PR DESCRIPTION
**Pull request by utkarsh530**
Fixes #283

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
No handling of the different dt?

File: src/DiffEqGPU.jl [hunk 1]
```diff
@@ -21,6 +21,45 @@ using Random
...
+
+@kernel function gpu_kernel(f, du, @Const(u),
+                            @Const(params::AbstractArray{ParamWrapper{P, T}}),
+                            @Const(t)) where {P, T}
+    i = @index(Global, Linear)
+    @inbounds p = params[i].params
+    @inbounds tspan = params[i].data
+    @views @inbounds f(du[:, i], u[:, i], p, t)
+    @inbounds for j in 1:size(du, 1)
+        du[j, i] = du[j, i] * (tspan[2] - tspan[1])
```
Comment (ChrisRackauckas) [inline src/DiffEqGPU.jl:46]:
```suggestion
        du[j, i] = u[j, i] * (tspan[2] - tspan[1])
```

File: src/DiffEqGPU.jl [hunk 2]
```diff
@@ -21,6 +21,45 @@ using Random
...
+@kernel function gpu_kernel_oop(f, du, @Const(u),
+                                @Const(params::AbstractArray{ParamWrapper{P, T}}),
+                                @Const(t)) where {P, T}
+    i = @index(Global, Linear)
+    @inbounds p = params[i].params
+    @inbounds tspan = params[i].data
+
+    @views @inbounds x = f(u[:, i], p, t)
+    @inbounds for j in 1:size(du, 1)
+        du[j, i] = x[j] * (tspan[2] - tspan[1])
```
Comment (ChrisRackauckas) [inline src/DiffEqGPU.jl:59]: no handling of the dt?

File: src/DiffEqGPU.jl [hunk 3]
```diff
@@ -755,20 +922,50 @@ function batch_solve(ensembleprob, alg,
...
-                                                           retcode = sol.retcode), i)[1]
-         for i in 1:length(probs)]
+
+        if !all(Base.Fix2((prob1, prob2) -> isequal(prob1.tspan, prob2.tspan),
+                          probs[1]),
+                probs)
+
+            # Requires prob.p to be isbits otherwise it wouldn't work with ParamWrapper
+            @assert all(prob -> isbits(prob.p), probs)
+            @info "Remaking the problem to normalize time span values..."
```
Comment (ChrisRackauckas) [inline src/DiffEqGPU.jl:933]: this shouldn't print every time.

File: src/DiffEqGPU.jl (hunk 2 above)
Comment (utkarsh530) [reply src/DiffEqGPU.jl:59]: No, not yet. I'll need to think about the logic concerning how to pass different dt's and separate different time-spans. I will follow-up in another PR.

File: README.md [hunk 4]
```diff
@@ -74,5 +74,22 @@ prob_func = (prob, i, repeat) -> remake(prob, p = (@SVector rand(Float32, 3)) .*
...
 
 @time sol = solve(monteprob, GPUTsit5(), EnsembleGPUKernel(CUDA.CUDABackend()),
-                  trajectories = 10_000, adaptive = false, dt = 0.1f0)
+    trajectories = 10_000, adaptive = false, dt = 0.1f0)
+```
+
+## Benchmarks
+
+Curious about our claims? See [https://github.com/utkarsh530/GPUODEBenchmarks](https://github.com/utkarsh530/GPUODEBenchmarks) for comparsion of our GPU solvers against CPUs and GPUs implementation in C++, JAX and PyTorch.
+
```
Comment (ChrisRackauckas) [inline README.md:83]: We should try to get this setup on the SciMLBenchmarks servers. @staticfloat @maleadt could we get help getting a GPU queue for benchmarking?

Comment (maleadt) [reply README.md:83]: There already is a `benchmark` queue that's used for CUDA.jl. Contains a single runner, albeit a fairly underpowered one.

Comment (ChrisRackauckas) [reply README.md:83]: We'd need some way to select which compute resource to do in each benchmark folder.

**Pull request merged**